### PR TITLE
Update tutorials

### PR DESCRIPTION
**Pull request by rmarquis**
* Add host to uvicorn command (required for Linux)
* Python 3.11 support
* Updated package versions
* Add export variable for cuda path
* Various fixes and enhancement

tested on Linux with updated service templates:
* extra_hosts section added in docker-compose
* SERVICE_URL in .env updated with host.docker.internal

### CODE REVIEW

File: docs/tutorials/create-a-service-to-rotate-an-image.md [hunk 1]
```diff
@@ -148,14 +148,18 @@ pip freeze --local --all > requirements-all.txt
 This will ensure that the same versions of the dependencies are installed on
 every machine if you ever share your code with someone else.
 
+As a last step, remove the `common-code @ git+https://github.com/swiss-ai-center/common-code.git@<commit>`
+line in `requirements-all.txt`, as it will conflict with the existing line in
+`requirements.txt` because of the explicit commit reference.
+
```
Comment (rmarquis) [inline docs/tutorials/create-a-service-to-rotate-an-image.md:154]: I am not sure I like this part (which is repeated in each tutorial, and what we did in production). While this is useful when we update the services running in production to always fetch the last version of common-code (that we do control), this effectively defeats the purpose of reproducibility in case of the update of the common-code repo, and seems counter-intuitive with best MLOps practices at first sight.

My suggestion for the tutorials would be not to remove the common-code line from the `requirements-all.txt` file, but to replace the generic common-code line in `requirements.txt` by the one generated in `requirements-all.txt`, to actually fix the common-code version as any other python dependency.

In a nutshell, with the current situation:

* Pro: we can fix all services with a new common-code version, just be regenerating the docker images and redeploying.
* Cons: We can also break services.

Proposed: fix the common-code version

* Pro: less risks of breakage
* Con: harder and longer to update services if we only need a common-code update.

Alternatively, explain both approaches in tutorials?

Comment (andrptrc) [reply docs/tutorials/create-a-service-to-rotate-an-image.md:154]: Perhaps the best way is to add both alternatives in the tutorial to let the developer choose if he wants his/her service to always be up to date and easily upgradable (for a swiss-ai-center dev for example) or if he/her wants it to be stable over time then he should keep the commit sha in the requirements-all

Comment (rmarquis) [reply docs/tutorials/create-a-service-to-rotate-an-image.md:168]: I've changed the tutorial to explictly request a "proper", fixed `common-code` version and added a note explaning the alternative. Feedback welcome!

File: docs/tutorials/create-a-service-to-detect-anomalies-using-a-model-built-from-scratch.md [hunk 2]
```diff
@@ -169,9 +169,27 @@ Create a freeze file to pin all dependencies to their current versions:
 pip freeze --local --all > requirements-all.txt
 ```
 
+Finally, copy the line
```
Comment (andrptrc) [inline docs/tutorials/create-a-service-to-detect-anomalies-using-a-model-built-from-scratch.md:172]: Wouldn't it be better to let the user choose if he wants to stay up to date or if he wants his service to be consistent over time. To do so tell him to either keep the line in requirements.txt or in requirements-all.txt. One for the updates and one for consistency ?

Comment (rmarquis) [reply docs/tutorials/create-a-service-to-detect-anomalies-using-a-model-built-from-scratch.md:172]: Is this better?:

```
The specific `common-code @ git+https://github.com/swiss-ai-center/common-code.git@<commit>`
line will conflict with the more general line in `requirements.txt` due to the explicit commit reference.

From there, you have two options:
* **Easier Update:** Remove the specific `common-code @ git+https://github.com/swiss-ai-center/common-code.git@<commit>` line from `requirements-all.txt`. This allows for easier updates of the common-code dependency without adjusting service dependencies.
* **Consistent Dependencies:** Remove the generic line in `requirements.txt` to keep dependencies consistent across machines and time. This will ensure that the same versions of the dependencies are installed on
every machine if you ever share your code with someone else.
```

Comment (andrptrc) [reply docs/tutorials/create-a-service-to-detect-anomalies-using-a-model-built-from-scratch.md:172]: Yes, this is more explicit !

**Review by andrptrc — APPROVED** [review]
